### PR TITLE
feat: add ceramic_one_info metric

### PR DESCRIPTION
**Pull request by nathanielc**
This adds an [Info](https://docs.rs/prometheus-client/latest/prometheus_client/metrics/info/struct.Info.html) metric to ceramic so that we have no basic build/version information about the process.

#### Review Note

 This change moves the one/src/main.rs to one/src/lib.rs so that we can export the Info type as needed. This makes the PR diff hard to read. But there are two separate commits, one where a `git mv` is used and another where the `main.rs` is added back in. This should mean reviewing the first commit only is what you really need to see.

### CODE REVIEW

File: one/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+        Self {
+            service_name: env!("CARGO_PKG_NAME").to_string(),
+            build: git_version::git_version!(
+                prefix = "git:",
+                cargo_prefix = "cargo:",
+                fallback = "unknown"
+            )
+            .to_string(),
+            version: env!("CARGO_PKG_VERSION").to_string(),
+            instance_id: names::Generator::default().next().unwrap(),
```
[the remark this answers is not in the document]
Comment (nathanielc) [reply one/src/lib.rs:541]: Correct once per startup.

File: one/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+    pub build: String,
+    /// Unique name generated for this invocation of the process.
+    pub instance_id: String,
+}
+
+impl Default for Info {
+    fn default() -> Self {
+        Self {
+            service_name: env!("CARGO_PKG_NAME").to_string(),
+            build: git_version::git_version!(
```
[the remark this answers is not in the document]
Comment (nathanielc) [reply one/src/lib.rs:534]: I like it, its not a hash of the binary, but a hash of the git commit. Makes it very clear which exact commit of the code is being run.

[the remark this answers is not in the document]
Comment (AaronGoldman) [reply one/src/lib.rs:534]: I get that this is the commit.
My question was whether we want to also hash the binary at startup and also publish the bin hash with this info.

[the remark this answers is not in the document]
Comment (nathanielc) [reply one/src/lib.rs:534]: ah I misread your comment.  Let me try that.

[the remark this answers is not in the document]
Comment (nathanielc) [reply one/src/lib.rs:540]: I got it working, turns out debug builds can be 1GB+ so I added logic to not hash for debug builds. Seems like a fair compromise as we likely only care about this for release builds.

**Review by AaronGoldman — APPROVED** [review]

**Pull request merged**